codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -284,35 +340,72 @@ private static ByteBuf encodeUnsubscribeMessage(
     }
 
     private static ByteBuf encodeSubAckMessage(
+            ChannelHandlerContext ctx,
             ByteBufAllocator byteBufAllocator,
             MqttSubAckMessage message) {
-        int variableHeaderBufferSize = 2;
+        MqttVersion mqttVersion = MqttCodecUtil.getMqttVersion(ctx);
+        ByteBuf propertiesBuf = encodePropertiesIfNeeded(mqttVersion,
```

Comment:
you will need to call `propertiesBuf.release()` later as you not transfer ownership